tests/ztest/include/ztest_assert.h

```diff
@@ -72,43 +84,50 @@ static inline void _zassert(int cond, const char *msg, const char *default_msg,
  * @param default_msg Message to print if @a cond is false
  */
 
-#define zassert(cond, msg, default_msg) \
-	_zassert(cond, msg ? msg : "", msg ? ("(" default_msg ")") : (default_msg), \
-		__FILE__, __LINE__, __func__)
+#define zassert(cond, default_msg, msg, ...) \
```

Comment:
zassert() is a public API, and we're reordering the parameters here.
any test case that uses zassert() is going to have to change, is there some other way we can do this?

Reply:
@andrewboie 

zassert() Macro has following comment :

"You probably don't need to call this macro directly. You should
instead use zassert_{condition} macros below."
I think it will called with zassert_(condition) only.

Reply:
but you are changing all zassert_{condition} as well :-)

But, I do not consider this a public API, this is a Zephyr internal testsuite and we should be able to adapt it to optimise testing. ztest.h is not even under include/ it is buried under tests/.

Reply:
@andrewboie  I dont see any other way as of now. can we able to merge same design ?